### ISSUE TITLE
Adding `updated_at` to `$dontVersionable` results in error when calling `diff()`

### ISSUE THREAD

**Issue by CamilleScholtz** [overtrue/laravel-versionable#56]
I don't really want to track `updated_at`, so I added it to the `$dontVersionable` variable in the model, however when calling the `diff()` method it gives me the following error:

```
Undefined array key "updated_at"
```

Full trace can be seen here: https://flareapp.io/share/Lm81QQJP#stack

**Comment by overtrue** [comment]
@mansoorkhan96 Please take a look. thanks.

**Comment by mansoorkhan96** [comment]
> I don't really want to track `updated_at`, so I added it to the `$dontVersionable` variable in the model, however when calling the `diff()` method it gives me the following error:
> 
> ```
> Undefined array key "updated_at"
> ```
> 
> Full trace can be seen here: https://flareapp.io/share/Lm81QQJP#stack

Please provide complete code examples to help us help you.

Code examples can help us reproduce the issue.

**Comment by mansoorkhan96** [comment]
@overtrue Seems like there are two issues with snapshot.

1) Initial version does not contain any attributes, so its always null
2) New versions are stored even though the $versionable attributes are not dirty

**Comment by mansoorkhan96** [comment]
@overtrue I think maybe this is happening because user aaded the DIFF strategy initially but then switched to Snapshot?

So one version which was created using DIFF might have a few attributes and the other one created with Snapshot would ofcourse have all the $versionable attributes. So thats where the diff() method complains about not having attributes in hand.

**Comment by mansoorkhan96** [comment]
I am able to reproduce.
Its the DIFF strategy which adds non $versionable attributes

**Comment by overtrue** [comment]
> @overtrue I think maybe this is happening because user aaded the DIFF strategy initially but then switched to Snapshot?
> 
> So one version which was created using DIFF might have a few attributes and the other one created with Snapshot would ofcourse have all the $versionable attributes. So thats where the diff() method complains about not having attributes in hand.

ok, may be we can ignore this, or add some notes in readme.

**Comment by mansoorkhan96** [comment]
@overtrue  I beleive there shouldnt be the DIFF strategy it should only be snapshot. which stores all attributes by default unless you specifiy $versionable.

**Comment by overtrue** [comment]
so, let's drop the strategy feature? ok?

**Comment by mansoorkhan96** [comment]
@overtrue yes that sounds good. plus a good mechanism to see if the changes are valid to create a new version.

plus some strictness maybe. throw an exception if user is using both properties i.e $versionable and $dontVersionfields. They should be using either of them?

If you want support both then maybe a way to get the intersection of both
